src/py/pyodide/code.py

```diff
@@ -25,11 +26,80 @@ def run_js(code: str, /) -> Any:
     return eval_(code)
 
 
+def _relaxed_call_sig(func):
+    from inspect import Parameter, signature
+
+    try:
+        sig = signature(func)
+    except (TypeError, ValueError):
+        return None
+    new_params = list(sig.parameters.values())
+    idx: int | None = -1
+    for idx, param in enumerate(new_params):
+        if param.kind in (Parameter.KEYWORD_ONLY, Parameter.VAR_KEYWORD):
+            break
+        if param.kind == Parameter.VAR_POSITIONAL:
+            idx = None
+            break
+    else:
+        idx += 1
+    if idx is not None:
+        new_params.insert(idx, Parameter("__var_positional", Parameter.VAR_POSITIONAL))
+
+    for param in new_params:
+        if param.kind == Parameter.KEYWORD_ONLY:
+            break
+    else:
+        new_params.append(Parameter("__var_keyword", Parameter.VAR_KEYWORD))
+    new_sig = sig.replace(parameters=new_params)
+    return new_sig
+
+
+@lru_cache
+def _relaxed_call_sig_cached(func):
+    return _relaxed_call_sig(func)
+
+
+def relaxed_wrap(func):
```

Comment:
It looks like the implementation of `relaxed_wrap` and `relaxed_call` are quite similar. Maybe we can define one using another, or extract common parts?

Reply:
Yeah having a helper like:
```py
def _do_call(func, sig, args, kwargs):
        bound = sig.bind(*args, **kwargs)
        bound.arguments.pop("__var_positional", None)
        bound.arguments.pop("__var_keyword", None)
        return func(*bound.args, **bound.kwargs)
```
would make sense.